core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -815,6 +815,39 @@ public static int bitLength(ByteString s) {
     return s.length() * 8;
   }
 
+  /** SQL BIT_GET(value, position) function. */
+  public static int bitGet(long value, int position) {
+    checkPosition(position, java.lang.Long.SIZE);
+    return (int) ((value >> position) & 1);
+  }
+
+  /** SQL BIT_GET(value, position) function. */
+  public static int bitGet(int value, int position) {
+    checkPosition(position, java.lang.Integer.SIZE);
+    return (value >> position) & 1;
```

Comment:
Is this expected behavior when it is the signed bit?

Reply:
I checked this is the expected behavior. This implementation is the same as Spark's and just returns the bit of the position, not specially considering the meaning of signed bit. I find that currently the related test case is missing in Spark, and I will add it in this PR.
Besides, the test cases for negative value are missing. I will add them to improve code coverage.